messaging/src/main/java/org/axonframework/serialization/json/JacksonConverter.java

```diff
@@ -16,124 +16,131 @@
+ * objects into and from a JSON format.
  * <p>
- * This converter provides robust and maintainable format handling for event payloads and other objects,
- * using Jackson for transformation between object and wire formats (such as JSON in String or byte[] form).
- * <p>
- * Intended for use with Axon Framework event storage and messaging, ensuring consistent payload
- * transformation between object and wire formats. Users may provide a custom-configured {@link ObjectMapper}
- * or use the default.
+ * Although the Jackson {@code Converter} requires classes to be compatible with this specific serializer, it provides
+ * much more compact serialization, while still being human-readable.
  *
+ * @author Allard Buijze
  * @author Mateusz Nowak
- * @since 5.0.0
- * TODO #3102 - Validate this Converter as part of #3102
+ * @author Steven van Beelen
+ * @since 2.2.0
  */
-public final class JacksonConverter implements Converter {
+public class JacksonConverter implements Converter {
 
     private static final Logger logger = LoggerFactory.getLogger(JacksonConverter.class);
 
     private final ObjectMapper objectMapper;
+    private final ChainingConverter converter;
 
     /**
-     * Constructs a JacksonFacultyEventConverter with a default ObjectMapper.
-     * The default mapper does not register any custom serializers or deserializers.
+     * Constructs a {@code JacksonConverter} with a default {@link ObjectMapper} that
+     * {@link ObjectMapper#findAndRegisterModules() finds and registers known modules}.
      */
     public JacksonConverter() {
         this(new ObjectMapper().findAndRegisterModules());
     }
 
     /**
-     * Constructs a JacksonFacultyEventConverter with a user-provided ObjectMapper.
+     * Constructs a {@code JacksonConverter} with the given {@code objectMapper}.
      *
-     * @param objectMapper The ObjectMapper to use for serialization/deserialization.
+     * @param objectMapper The mapper used to convert objects into and from a JSON format.
      */
-    public JacksonConverter(ObjectMapper objectMapper) {
-        this.objectMapper = objectMapper;
+    public JacksonConverter(@Nonnull ObjectMapper objectMapper) {
+        this.objectMapper = Objects.requireNonNull(objectMapper, "The ObjectMapper may not be null.");
+        this.objectMapper.registerModule(new JavaTimeModule());
+        // TODO The Converter used to be configurable for the JacksonSerializer. I don't think we need that anymore. Thoughts?
+        this.converter = new ChainingConverter();
+        this.converter.registerConverter(new JsonNodeToByteArrayConverter(this.objectMapper));
+        this.converter.registerConverter(new ByteArrayToJsonNodeConverter(this.objectMapper));
+        this.converter.registerConverter(new JsonNodeToObjectNodeConverter());
+        this.converter.registerConverter(new ObjectNodeToJsonNodeConverter());
```

Comment:
I prefer to keep it internal and allow configurable if really needed in the future - we will see the usages.